src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -155,6 +186,7 @@ create_reference_table(PG_FUNCTION_ARGS)
 	Oid relationId = PG_GETARG_OID(0);
 	int shardCount = 1;
 	AttrNumber firstColumnAttrNumber = 1;
+	char *colocateWithTableName = "default";
```

Comment:
Here, we force reference tables to use default colocation group. However there can be another colocation group which has one shard and replication factor = node count. And if this colocation group is created before first reference table, it will be default colocation group for that setting. Therefore we will create reference tables in wrong colocation group. This will also probably create some problems during shard rebalance.

Also even if a reference table is created first, therefore correct colocation setting become default, user may change default colocation group to something else, which cause subsequent reference table creations to wrong colocation group.

I think reference tables should have their own reserved colocation id. @onderkalaci what do you think about that?